src/datasets-string.c

```diff
@@ -73,6 +73,24 @@ int StringSet(void *dst, void *src)
     return 0;
 }
 
+int StringJsonSet(void *dst, void *src)
+{
+    StringType *src_s = src;
+    StringType *dst_s = dst;
+    SCLogDebug("dst %p src %p, src_s->ptr %p src_s->len %u", dst, src, src_s->ptr, src_s->len);
+
+    dst_s->len = src_s->len;
+    dst_s->ptr = SCMalloc(dst_s->len);
+    BUG_ON(dst_s->ptr == NULL);
```

Comment:
nit: Should we have `BUG_ON` for malloc failure?

Reply:
I'm unsure. You're right the pattern exists at other places too. Usually, I see the pattern to return with an error code when malloc is unsuccessful. Let's let it be for now.